### PR TITLE
Adds pre-trained re-weighting to SVC classifiers and tests to demonstrate the improved predict_proba performance.

### PR DESCRIPTION
**Pull request by halfak**
See the nonsense this fixes here: http://nbviewer.ipython.org/github/halfak/Objective-Revision-Evaluation-Service/blob/ipython/ipython/Wat%20predict_proba.ipynb

See the relevant bug report here: https://github.com/halfak/Revision-Scoring/issues/45

### CODE REVIEW

File: revscoring/scorers/svc.py [hunk 1]
```diff
@@ -131,6 +125,21 @@ def test(self, values_labels, comparison_class="auto"):
             'auc': auc(fpr, tpr)
         }
     
+    def _balance_labels(self, values_labels):
+        groups = defaultdict(lambda: [])
```
Comment (he7d3r) [inline revscoring/scorers/svc.py]: Just for curiosity, is this line equivalent to `groups = defaultdict(list)`?

File: revscoring/scorers/svc.py [hunk 2]
```diff
@@ -131,6 +125,21 @@ def test(self, values_labels, comparison_class="auto"):
...
+        for feature_values, label in values_labels:
+            groups[label].append(feature_values)
+        
+        max_label_n = max(len(groups[label]) for label in groups)
+        
+        new_values_labels = []
+        for label in groups:
+            new_values_labels.extend((random.choice(groups[label]), label)
+                                      for i in range(max_label_n))
```
Comment (he7d3r) [inline revscoring/scorers/svc.py]: What does this loop do? Does it duplicate a few random examples of the class(es) with the a small number of representatives until we have all classes with the same size? Or something else?
Maybe add a short comment explaining the idea?

File: revscoring/scorers/tests/test_svc.py [hunk 3]
```diff
@@ -45,9 +45,13 @@ def train_score(model):
...
     
     model.train(train_set)
-    score_doc = next(model.score([(-1,-2)]))
+    score_doc = next(model.score([(-.3,-.3)]))
     
-    eq_(score_doc['prediction'], False)
+    eq_(score_doc['prediction'], True)
+    print(score_doc)
+    assert score_doc['probability'][True] > 0.5, \
+           "Probability of True {0} is not > 0.5" \
```
Comment (he7d3r) [inline revscoring/scorers/tests/test_svc.py]: Just to be sure: is this supposed to be always ```> 0.5``` or ```>= 0.5```?

File: revscoring/scorers/tests/test_svc.py [hunk 4]
```diff
@@ -45,9 +45,13 @@ def train_score(model):
     test_set = observations[mid:]
     
     model.train(train_set)
-    score_doc = next(model.score([(-1,-2)]))
+    score_doc = next(model.score([(-.3,-.3)]))
     
-    eq_(score_doc['prediction'], False)
+    eq_(score_doc['prediction'], True)
+    print(score_doc)
```
Comment (he7d3r) [inline revscoring/scorers/tests/test_svc.py]: Stray `print`?

File: revscoring/scorers/svc.py (hunk 1 above)
Comment (halfak) [inline revscoring/scorers/svc.py]: Yes.  I'll fix that in a follow-up.

File: revscoring/scorers/svc.py (hunk 2 above)
Comment (halfak) [inline revscoring/scorers/svc.py]: Will be included in a follow-up.

File: revscoring/scorers/tests/test_svc.py (hunk 3 above)
Comment (halfak) [inline revscoring/scorers/tests/test_svc.py]: Shouldn't matter, but I don't mind making the change.

**Comment by halfak** [comment]
@he7d3r, please have another look.  :)

**Pull request merged**